api/v1/verticadb_webhook.go

```diff
@@ -320,6 +322,43 @@ func (v *VerticaDB) validateEndpoint(allErrs field.ErrorList) field.ErrorList {
 	return allErrs
 }
 
+func (v *VerticaDB) hasValidSvcName(allErrs field.ErrorList) field.ErrorList {
+	vdbName := v.ObjectMeta.Name
+	// vdb name will be used as is for headless svc name,
+	// thus match regex for service name (DNS-1035 label)
+	r := regexp.MustCompile(`^[a-z]([\-a-z0-9]{0,61}[a-z0-9])?$`)
+	if !r.MatchString(vdbName) {
```

Comment:
You do not need to validate vdb name. Resource names(ObjectMeta.Name) are already checked against a set of [rules](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names)

Reply:
No you can try yourself, "1v-" is going to fail.

Reply:
Object.meta is already being validated against this regex `'[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')`

Reply:
In that case create a separate function to validate vdb.